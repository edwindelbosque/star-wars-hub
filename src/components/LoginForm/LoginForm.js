import React, { Component } from 'react';
import './LoginForm.scss';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

class LoginForm extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      quote: '',
      skillLevel: ''
    }
  }

  handleChange = (e) => {
    e.preventDefault();
    this.setState({
      [e.target.name]: e.target.value
    })
  }

  handleClick = (event) => {
    const { name, quote, skillLevel } = this.state;
    if (!name || !quote || !skillLevel) {
      event.preventDefault()
    } else {
      this.props.addUserInfo(this.state)
    }
  }

  render() {
    const { name, quote, skillLevel } = this.state
    return (
      <form>
        <h1>STAR WARS HUB</h1>
        <h4>Join the force</h4>
        <label>Name</label>
        <input
          type='text'
          name='name'
          className='input-1'
          value={name}
          placeholder='Enter name here'
          onChange={(e) => this.handleChange(e)}
        />
        <label>Quote</label>
        <input
          type='text'
          name='quote'
          className='input-2'
          value={quote}
          placeholder='Enter your favorite Star Wars quote'
          onChange={(e) => this.handleChange(e)}
        />
        <label>Skill Level</label>
        <select
          name="skillLevel"
          onChange={(e) => this.handleChange(e)}
        >
          <option value=''>Select Skill Level</option>
          <option value='Jedi Initiate'>Jedi Initiate</option>
          <option value='Jedi Knight'>Jedi Knight</option>
          <option value='Jedi Master'>Jedi Master</option>
        </select>
        <div>
          <Link to="/movies" >
            <button
              type='button'
              onClick={(event) => this.handleClick(event)}>
              Submit
            </button>
          </Link>
        </div>
      </form >
    )
  }
}

LoginForm.propTypes = {
  name: PropTypes.string,
  quote: PropTypes.string,
  skillLevel: PropTypes.string
};

export default LoginForm