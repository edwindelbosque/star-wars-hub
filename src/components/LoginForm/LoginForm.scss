h1 {
  font-size: 70px;
  color: rgb(255, 225, 0);
}

h4 {
  width: 100%;
  text-align: center;
  padding-right: 10px;
}

form {
  width: 500px;
  height: 570px;
  margin: auto;
  margin-top: 60px;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: flex-start;
  background-color: #383838;
  background: linear-gradient(#55729ee3, #3838383b);
  padding: 40px;
  border: 1px solid rgb(0, 136, 255);
  border-bottom: 1px solid rgb(72, 36, 255);
  border-top: 1px solid #5bc6c9e3;
  border-radius: 5px;
  box-shadow: 0 0 100px rgba(126, 188, 255, 0.365);
  transition: .4s;

  &:hover {
    box-shadow: 0 0 100px rgba(126, 171, 255, 0.509);
    transition: .4s;
  }

  label {
    font-size: 16px;
    color: white;
    font-weight: 600;
    margin-left: 30px;
    margin-bottom: -12px;
  }

  input {
    height: 30px;
    width: 110%;
    border-radius: 0 100px 100px 0;
    font-size: 16px;
    margin-left: -30px;
    border: none;
    border-left: 60px solid rgb(19, 19, 19);
    padding-left: 10px;
    background: rgb(255, 255, 255);
    transition: .3s;

    &:focus {
     outline: none; 
     box-shadow: 0 0 30px 0 #FF0000;
     transition: .25s;
    }
  }

  .input-1 {
    &:focus {
      box-shadow: 0 0 40px 0 #FF0000;
      background-color: rgb(255, 224, 224);
      transition: .25s;
     }

    &:active {
      box-shadow: 0 0 10px 0 #FF0000;
      background-color: rgb(255, 255, 255);
      transition: .25s;
    }
  }

  .input-2 {
    &:focus {
      box-shadow: 0 0 40px 0 rgb(61, 255, 43);
      background-color: rgb(224, 244, 255);
      transition: .25s;
     }

     &:active {
      box-shadow: 0 0 10px 0 rgb(61, 255, 43);
      background-color: rgb(255, 255, 255);
      transition: .25s;
    }
  }
  
  ::placeholder {
    color: rgb(174, 174, 174);
  }
  
  select {
    height: 30px;
    font-size: 16px;
    width: 150px;
    margin-left: 30px;
    background-color: white;
  }

  div {
    margin-top: 30px;
    width: 100%;
  }

  button {
    border: none;
    border-radius: 5px;
    background-color: rgb(255, 217, 0);
    width: 100px;
    height: 40px;
    font-size: 18px;
    font-weight: 600;
    transition: .25s;
    box-shadow: 0 2px 3px rgba(0, 0, 0, 0.557);

    &:hover {
      background-color: rgb(255, 242, 168);
      transition: .25s;
      box-shadow: 0 0 10px 0 rgb(255, 229, 32);
    }

    &:active {
      background-color: rgb(255, 255, 255);
      transition: .1s;
      box-shadow: 0 0 20px 0 rgb(255, 229, 32);
    }

    &:focus {
      outline: none;
    }
  }
}

.error-message {
  border: 2px solid rgb(255, 62, 62);
}