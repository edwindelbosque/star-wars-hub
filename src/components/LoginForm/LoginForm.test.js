import React from 'react';
import { shallow } from 'enzyme';
import LoginForm from './LoginForm';

describe('LoginForm', () => {
  const addUserInfo = jest.fn();
  it('component should match the snapshot', () => {
    const wrapper = shallow(
      <LoginForm addUserInfo={addUserInfo} />
    )
    expect(wrapper).toMatchSnapshot();
  });

  it('should update quote name with new name onChange input', () => {
    // Setup
    const wrapper = shallow(<LoginForm />);
    const mockEvent = { preventDefault() { }, target: { name: 'name', value: 'A' } };
    const expected = 'A';

    // Execution
    wrapper.instance().handleChange(mockEvent);

    // Expectation
    expect(wrapper.state('name')).toEqual(expected);
  });

  it('should update quote state with new quote onChange input', () => {
    // Setup
    const wrapper = shallow(<LoginForm />);
    const expected = 'May the force be with you.';
    const mockEvent = { preventDefault() { }, target: { name: 'quote', value: expected } };

    // Execution
    wrapper.instance().handleChange(mockEvent);

    // Expectation
    expect(wrapper.state('quote')).toEqual(expected);
  });

  it('should update skillLevel state with new skillLevel onChange select', () => {
    // Setup
    const wrapper = shallow(<LoginForm />);
    const expected = 'Expert';
    const mockEvent = { preventDefault() { }, target: { name: 'skillLevel', value: expected } };

    // Execution
    wrapper.instance().handleChange(mockEvent);

    // Expectation
    expect(wrapper.state('skillLevel')).toEqual(expected);
  });

  it('should call the handleClick prop with the forms information when clicked', () => {
    // Setup
    const wrapper = shallow(<LoginForm />);
    wrapper.instance().handleClick = jest.fn();

    // Execution
    wrapper.find('button').simulate('click');

    // Expectation
    expect(wrapper.instance().handleClick).toHaveBeenCalled();
  });

  it('should call the onChange method when user types', () => {
    // Setup
    const wrapper = shallow(<LoginForm />);
    const NameEvent = { target: { name: 'name', value: 'A' } };
    const QuoteEvent = { target: { name: 'quote', value: 'B' } };
    const SkillEvent = { target: { name: 'quote', value: 'B' } };
    wrapper.instance().handleChange = jest.fn();

    // Execution
    wrapper.find('[name="name"]').simulate('change', NameEvent);
    wrapper.find('[name="quote"]').simulate('change', QuoteEvent);
    wrapper.find('[name="skillLevel"]').simulate('change', SkillEvent);

    // Expectation
    expect(wrapper.instance().handleChange).toHaveBeenCalledWith(NameEvent);
    expect(wrapper.instance().handleChange).toHaveBeenCalledWith(QuoteEvent);
    expect(wrapper.instance().handleChange).toHaveBeenCalledWith(SkillEvent);
  });

})

