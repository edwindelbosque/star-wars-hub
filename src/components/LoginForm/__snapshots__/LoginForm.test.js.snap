// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component should match the snapshot 1`] = `
<form>
  <h1>
    SWAPI Trivia
  </h1>
  <h4>
    Join the force
  </h4>
  <label>
    Name
  </label>
  <input
    className="input-1"
    name="name"
    onChange={[Function]}
    placeholder="Enter name here"
    type="text"
    value=""
  />
  <label>
    Quote
  </label>
  <input
    className="input-2"
    name="quote"
    onChange={[Function]}
    placeholder="Enter your favorite Star Wars quote"
    type="text"
    value=""
  />
  <label>
    Skill Level
  </label>
  <select
    name="skillLevel"
    onChange={[Function]}
  >
    <option
      value=""
    >
      Select Skill Level
    </option>
    <option
      value="Novice"
    >
      Novice
    </option>
    <option
      value="Intermediate"
    >
      Intermediate
    </option>
    <option
      value="Expert"
    >
      Expert
    </option>
  </select>
  <div>
    <Link
      to="/movies"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Submit
      </button>
    </Link>
  </div>
</form>
`;
